internal/controller/changetransferpolicy_controller_test.go

```diff
@@ -202,7 +206,80 @@ var _ = Describe("ChangeTransferPolicy Controller", func() {
 				}
 				err := k8sClient.Get(ctx, typeNamespacedNamePR, &pr)
 				g.Expect(errors.IsNotFound(err)).To(BeTrue())
-			})
+			}, EventuallyTimeout).Should(Succeed())
+		})
+
+		It("webhook should modify annotation", func() {
+			webhookPort := WebhookReceiverPort + GinkgoParallelProcess()
+			webhookURL := fmt.Sprintf("http://localhost:%d/", webhookPort)
+
+			name, scmSecret, scmProvider, gitRepo, _, changeTransferPolicy := changeTransferPolicyResources(ctx, "ctp-webhook", "default")
+
+			typeNamespacedName := types.NamespacedName{
+				Name:      name,
+				Namespace: "default", // TODO(user):Modify as needed
+			}
+
+			changeTransferPolicy.Spec.ProposedBranch = "environment/development-next"
+			changeTransferPolicy.Spec.ActiveBranch = "environment/development"
+			// We set auto merge too false to avoid the PR being merged automatically so we can run checks on it
+			changeTransferPolicy.Spec.AutoMerge = ptr.To(false)
+
+			Expect(k8sClient.Create(ctx, scmSecret)).To(Succeed())
+			Expect(k8sClient.Create(ctx, scmProvider)).To(Succeed())
+			Expect(k8sClient.Create(ctx, gitRepo)).To(Succeed())
+			Expect(k8sClient.Create(ctx, changeTransferPolicy)).To(Succeed())
+
+			gitPath, err := os.MkdirTemp("", "*")
```

Comment:
Need cleanup?

Reply:
eh... maybe its in a temp dir will cleanup on reboots :P

There are other test based temp dirs not cleaned up so will do that in different PR